src/include/mpir_ext.h.in

```diff
@@ -68,9 +68,10 @@ extern int MPIR_Ext_dbg_romio_terse_enabled;
 extern int MPIR_Ext_dbg_romio_typical_enabled;
 extern int MPIR_Ext_dbg_romio_verbose_enabled;
 
-/* to be called early by ROMIO's initialization process in order to setup init-time
+/* to be called early by MPIR_Init_thread when ROMIO is included in order to setup init-time
```

Comment:
Hmm, I'm not sure we can move this initialization inside MPICH. That would break users such as Open MPI, no?

Reply:
Hehe, "break" is a strong word. Yes, for other users, they'll need to update their code to adapt to this change. A library that requires init/finalize is not a hacky idea, right? Users just need to know that they need to call init/finalize. I think it is better than the current lazy init approach, which is obscure and non-expandable. From what I see, the current way other users to use ROMIO is a hacky process anyway -- requiring their own `mpir_ext.h`. FWIW, after this patch, ROMIO is nearly independent from the rest of the MPICH, and with another patch, we probably can get rid of `mpir_ext.h` and make ROMIO independent package, which should greatly simplify how other users use it.

As for the impact, does open-mpi directly take romio from mpich repository?

Reply:
I guess we can simply use static mutex initialization. Let me try that.

Reply:
> 
> 
> I guess we can simply use static mutex initialization. Let me try that.

Oh, I see why we can't. We are using MPL wrappers. I guess we can't assume pthread mutex is always available even though in practice it does.  So --

* Option 1. abandon MPL wrappers and directly use pthread_mutex for `romio_mutex`, adding hard dependency on pthread even for THREAD_SINGLE build

* Option 2. use MPL wrappers. Then we can't do lazy init.

Reply:
> Hehe, "break" is a strong word. Yes, for other users, they'll need to update their code to adapt to this change. A library that requires init/finalize is not a hacky idea, right?

It is a backwards compatibility "break". 

> Users just need to know that they need to call init/finalize. 

Then this needs additional documentation. Still, I think limiting the scope of changes to get this merged would be a good idea.

> As for the impact, does open-mpi directly take romio from mpich repository?

Yes.

Reply:
Moved the mutex init to lazy init. There is no place to call finalize, so we'll just leave it like that. Should be ok for pthread mutex, but it's a small leak for argobots.

Reply:
Thanks. It's not perfect, but I think it should work :).